server/parsec/components/memory/shamir.py

```diff
@@ -0,0 +1,47 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+from parsec._parsec import DeviceID, OrganizationID, UserID, VerifyKey, authenticated_cmds
+from parsec.components.events import EventBus
+from parsec.components.memory.datamodel import (
+    MemoryDatamodel,
+    MemoryShamirSetup,
+)
+from parsec.components.shamir import (
+    BaseShamirComponent,
+    VerifyCertificatesBadOutcome,
+    verify_certificates,
+)
+
+
+class MemoryShamirComponent(BaseShamirComponent):
+    def __init__(self, data: MemoryDatamodel, event_bus: EventBus) -> None:
+        super().__init__()
+        self._data = data
+        self._event_bus = event_bus
+
+    async def remove_recovery_setup(
+        self,
+        organization_id: OrganizationID,
+        author: UserID,
+    ) -> None:
+        self._data.organizations[organization_id].shamir_setup.pop(author)
+
+    async def add_recovery_setup(
```

Comment:
Some sanity_check is probably required before calling `verify_certificates`: the organization exists and is not expired? author exists and is not revoked? 
See for example https://github.com/Scille/parsec-cloud/blob/53642307a06c6eddf2e1de42226d750cacfe7dc5/server/parsec/components/memory/realm.py#L79-L92

BTW there is an ongoing discussion about these checks (error-prone, duplicated code): #7119